main.go

```diff
@@ -178,33 +179,25 @@ func HandleRun(w http.ResponseWriter, req *http.Request) {
 	if err != nil {
 		log.Println(err)
 	}
-
 }
 
-func HandleCancel(w http.ResponseWriter, req *http.Request) {
-	log.Print("Received cancel")
-	runCancel()
-}
-
-// currently bash complete only, wrapper around "Run"
-func Complete(req io.Reader) (string, error) {
-	buf, err := ioutil.ReadAll(req)
+func HandleComplete(w http.ResponseWriter, req *http.Request) {
+	output, err := Exec(shell.Complete, req.Body)
```

Comment:
You should just call shell.Complete() here directly. You don't need to reuse any of the behavior in Run()/Exec(), such as setting up PTYs or parsing escape sequences.